### DISCUSSION TITLE
What does "Allow Discharge" option do differently?

### DISCUSSION THREAD

**Discussion by mikebore** [AppHouseKitchen/AlDente-Battery_Care_and_Monitoring#169] [Q&A]
Hi, the description of the "Allow Discharge" option says:

_"Allow Discharge - When this feature is activated, your MacBook will run on battery until it reaches the desired percentage. Afterwards, it will stay at the desired battery percentage and use the power adapter again"_

Doesn't it do the same thing if "Allow Discharge" is not checked?

What does "Allow Discharge" do differently?

Using M1 MacBook Air in case this makes a difference.

Thanks for great tool.

**Comment by mikebore** [comment]
After some testing, including using a USB-C in line power meter to show what is actually going into the machine, I think I can answer my own question. 

As expected the Allow Discharge setting only makes a difference while the battery level is **ABOVE** the max set by AlDente:

(a) with "Allow Discharge" OFF, the machine still takes power from the charger and the battery level is very slow to drop.

(b) with "Allow Discharge" ON, the machine does not take any power from the charger, only from the battery, and battery level drops rapidly.

I started with the battery at 100% and with AlDente set to 80% max. I set the Mac to play long home movies.

With Allow Discharge OFF, the meter showed the machine was taking 0.3A at 20V and the battery level took two hours to drop to 99.5%. 

With Allow Discharge ON, the meter showed the machine was taking 0.01A at 20V and the battery level dropped rapidly. When charge level dropped 80%, it went back to 0.3A at 20V, enough to run the machine but less than it would be if charging the battery as well.

With hindsight this is probably all rather obvious but I was having a bit of trouble getting my head round it!

**Reply by JeanMichelForgeas** [reply]
Thanks a lot for explanations
